### ISSUE TITLE
Controller not listening to Pub/Sub commands

### ISSUE THREAD

**Issue by abaumann** [GoogleCloudPlatform/dataflow-opinion-analysis#4]
This is a great and well documented example - much appreciated!  

I am having trouble however with the Run a verification job step.  When I publish the `command=start_gcs_import` message to the `indexercommands` topic I do not get a new input job created.  I tried out the cron jobs to see if those were working, and I can see a new job created for `startstatscalc` only - not the others.  I feel like I must have messed up some config step along the way.

Do you have tips for how to debug what is happening?  I am not seeing any sort of logging to help me debug, but that's where I'd think to look first.  It feels like the controller must not be listening for the topics correctly though...

Thanks!

**Comment by datancoffee** [comment]
yes, sorry, the controller pipeline has not yet been updated (after a recent upgrade from a pre 2.x SDK to a 2.2 SDK). 

For the time being, launch the IndexerPipeline directly using the example in the Release Notes for version 0.6.4 https://github.com/GoogleCloudPlatform/dataflow-opinion-analysis/releases/tag/v0.6.4

e.g. using

mvn compile exec:java 
-Dexec.mainClass=com.google.cloud.dataflow.examples.opinionanalysis.IndexerPipeline 
-Dexec.args="--project=$PROJECT_ID 
...


I will make a note in README not to run the controller pipeline but instead to start the IndexerPipeline

**Comment by datancoffee** [comment]
Just insert something like this into the shell script (and make sure the $ variables are set)
(this statement is from the release notes) 

mvn compile exec:java 
-Dexec.mainClass=com.google.cloud.dataflow.examples.opinionanalysis.IndexerPipeline 
-Dexec.args="--project=$PROJECT_ID 
--runner=DataflowRunner 
--maxNumWorkers=10 
--workerMachineType=n1-standard-2 
--stagingLocation=gs://$GCS_BUCKET/staging/ 
--tempLocation=gs://$GCS_BUCKET/temp/ 
--streaming=false 
--autoscalingAlgorithm=THROUGHPUT_BASED 
--bigQueryDataset=opinions 
--writeTruncate=true 
--processedUrlHistorySec=130000 
--wrSocialCountHistoryWindowSec=610000 
--ratioEnrichWithCNLP=0 
--sourceRecordFile=true 
--inputFile=gs://$GCS_BUCKET/input/*.txt 
--indexAsShorttext=false
"

**Comment by abaumann** [comment]
Oh actually that worked - I thought I looked closely enough to see the flags were the same there as the existing template, but I guess not, thanks!

**Comment by datancoffee** [comment]
glad to hear, and sorry for the issue with the controller. I meant to fix it earlier

**Comment by abaumann** [comment]
no problem - I just wanted a baseline of something working so I can start to modify from here to handle our use case, so now that I've seen this working, I'm good.  Only other issue I noticed by the way is the instructions to run: 
```
SELECT * FROM opinions.sentiment 
ORDER BY DocumentTime DESC
LIMIT 100
```

Since this contains repeated fields we get an error like this"
`Cannot query the cross product of repeated fields Signals and Tags.GoodAsTopic`

I really hate that select * doesn't do some default flattening in BQ, and I didn't do the work to come up with a query and just used the preview pane instead.

Thanks!

**Comment by datancoffee** [comment]
Uncheck the "Use Legacy SQL" checkbox so that BigQuery uses Standard SQL for that query. You can do that by going to "Show Options"

**Comment by datancoffee** [comment]
But, good point, I will add the 
#standardSQL
tag to the query, so that it is easier to use this sample.

**Comment by abaumann** [comment]
:fireworks: i did not know about unchecking Legacy SQL being a fix to this - thanks for that tip! :fireworks:
